### ISSUE TITLE
aws-cfn-bootstrap error in ubuntu16.04 pcluster builds

### ISSUE THREAD

**Issue by amrragab8080** [aws/aws-parallelcluster#1157]
**Environment:**
 - aws-parallelcluster-2.4
 - OS: ubuntu16.04

**Bug description and how to reproduce:**
In the parallelcluster cookbook processing and install aws-cfn-bootstrap there is currently a regression with python-daemon in py3 which causes cfn-init to fail.

```
+ cfn-signal --exit-code=1 '--reason=Failed to run cfn-init. If --norollback was specified, check /var/log/cfn-init.log and /var/log/cloud-init-output.log.' --stack=parallelcluster-default2 --resource=MasterServer --region=us-west-2
Traceback (most recent call last):
  File "/usr/local/bin/cfn-signal", line 4, in <module>
    __import__('pkg_resources').run_script('aws-cfn-bootstrap==1.4', 'cfn-signal')
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 3126, in <module>
    @_call_aside
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 3110, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 3139, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 581, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 898, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 784, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'python-daemon<2.0,>=1.5.2' distribution was not found and is required by aws-cfn-bootstrap
+ exit 1
```
The error is in the dependency of python-daemon errors earlier in the cookbook.
```
Running python-daemon-1.6.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-53mq_2rq/python-daemon-1.6.1/egg-dist-tmp-xlltty5a
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/usr/local/lib/python3.5/dist-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/local/lib/python3.5/dist-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/usr/local/lib/python3.5/dist-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-53mq_2rq/python-daemon-1.6.1/setup.py", line 24, in <module>
  File "/tmp/easy_install-53mq_2rq/python-daemon-1.6.1/daemon/__init__.py", line 42, in <module>
  File "/tmp/easy_install-53mq_2rq/python-daemon-1.6.1/daemon/version/__init__.py", line 17, in <module>
  File "/tmp/easy_install-53mq_2rq/python-daemon-1.6.1/daemon/version/version_info.py", line 20
    print 'revision: %(revno)s' % version_info
                              ^
SyntaxError: invalid syntax
```
Linking a similar issue. https://github.com/spotify/luigi/issues/2632

**Comment by sean-smith** [comment]
Thanks for reporting this @amrragab8080 we're looking into it

**Comment by amrragab8080** [comment]
Adding more detail - The AMI being used is here a processed pcluster createami build of the Deep Learning AMI Ubuntu Base image. https://aws.amazon.com/marketplace/pp/B077GCZ4GR#pdp-pricing

**Comment by demartinofra** [comment]
As part of [ParallelCluster 2.4.1](https://github.com/aws/aws-parallelcluster/releases/tag/v2.4.1) we released a series of enhancements that allow the cluster daemons and cookbook python dependencies to be installed in isolated virtualenvs. Could you try to create a new custom ami starting from those released with the 2.4.1 and see if the issue still occurs? Thank you
